clippy_lints/src/loops/mod.rs

```diff
@@ -0,0 +1,669 @@
+            while_let_loop::check_while_let_loop(cx, expr, block);
+        }
+
+        while_let_on_iterator::check_while_let_on_iterator(cx, expr);
+
+        if let Some((cond, body)) = higher::while_loop(&expr) {
+            while_immutable_condition::check_infinite_loop(cx, cond, body);
+        }
+
+        needless_collect::check_needless_collect(expr, cx);
+    }
+}
+
+fn check_for_loop<'tcx>(
+    cx: &LateContext<'tcx>,
+    pat: &'tcx Pat<'_>,
+    arg: &'tcx Expr<'_>,
+    body: &'tcx Expr<'_>,
+    expr: &'tcx Expr<'_>,
+    span: Span,
+) {
+    let is_manual_memcpy_triggered = manual_memcpy::detect_manual_memcpy(cx, pat, arg, body, expr);
+    if !is_manual_memcpy_triggered {
+        needless_range_loop::check_for_loop_range(cx, pat, arg, body, expr);
+        explicit_counter_loop::check_for_loop_explicit_counter(cx, pat, arg, body, expr);
+    }
+    check_for_loop_arg(cx, pat, arg, expr);
+    for_kv_map::check_for_loop_over_map_kv(cx, pat, arg, body, expr);
+    mut_range_bound::check_for_mut_range_bound(cx, arg, body);
+    single_element_loop::check_for_single_element_loop(cx, pat, arg, body, expr);
+    same_item_push::detect_same_item_push(cx, pat, arg, body, expr);
+    manual_flatten::check_manual_flatten(cx, pat, arg, body, span);
+}
+
+fn check_for_loop_arg(cx: &LateContext<'_>, pat: &Pat<'_>, arg: &Expr<'_>, expr: &Expr<'_>) {
+    let mut next_loop_linted = false; // whether or not ITER_NEXT_LOOP lint was used
+    if let ExprKind::MethodCall(ref method, _, ref args, _) = arg.kind {
+        // just the receiver, no arguments
+        if args.len() == 1 {
+            let method_name = &*method.ident.as_str();
+            // check for looping over x.iter() or x.iter_mut(), could use &x or &mut x
+            if method_name == "iter" || method_name == "iter_mut" {
+                if is_ref_iterable_type(cx, &args[0]) {
+                    explicit_iter_loop::lint_iter_method(cx, args, arg, method_name);
+                }
+            } else if method_name == "into_iter" && match_trait_method(cx, arg, &paths::INTO_ITERATOR) {
+                let receiver_ty = cx.typeck_results().expr_ty(&args[0]);
+                let receiver_ty_adjusted = cx.typeck_results().expr_ty_adjusted(&args[0]);
+                if TyS::same_type(receiver_ty, receiver_ty_adjusted) {
+                    explicit_into_iter_loop::check_explicit_into_iter_loop(cx, args, arg);
+                } else {
+                    let ref_receiver_ty = cx.tcx.mk_ref(
+                        cx.tcx.lifetimes.re_erased,
+                        ty::TypeAndMut {
+                            ty: receiver_ty,
+                            mutbl: Mutability::Not,
+                        },
+                    );
+                    if TyS::same_type(receiver_ty_adjusted, ref_receiver_ty) {
+                        explicit_iter_loop::lint_iter_method(cx, args, arg, method_name)
+                    }
```

Comment:
ditto.
The implemetation of [`metods/mod.rs`](https://github.com/rust-lang/rust-clippy/blob/928e72dd10749875cbd412f74bfbfd7765dbcd8a/clippy_lints/src/methods/mod.rs#L15) would be great help to grab the idea.
```suggestion
                explicit_into_iter_loop::check_explicit_into_iter_loop(cx, args, arg);
                explicit_iter_loop::check_explicit_into_iter_loop(cx, args, arg);
```